pkg/controllers/clusterresourceplacement/controller_test.go

```diff
@@ -1126,38 +1127,68 @@ func secretResourceContentForTest(t *testing.T) *fleetv1beta1.ResourceContent {
+	serviceResourceContent := *serviceResourceContentForTest(t)
+	// test deployment 390 bytes in size.
+	deploymentResourceContent := *deploymentResourceContentForTest(t)
+	// test secret is 152 bytes in size.
+	secretResourceContent := *secretResourceContentForTest(t)
+	resourceSnapshotSpecWithSingleResource := &fleetv1beta1.ResourceSnapshotSpec{
+		SelectedResources: []fleetv1beta1.ResourceContent{serviceResourceContent},
 	}
-	jsonBytes, err := json.Marshal(resourceSnapshotSpecA)
+	jsonBytes, err := json.Marshal(resourceSnapshotSpecWithSingleResource)
 	if err != nil {
-		t.Fatalf("failed to create the resourceSnapshotSpecA hash: %v", err)
+		t.Fatalf("failed to create the resourceSnapshotSpecWithSingleResource hash: %v", err)
 	}
-	resourceSnapshotAHash := fmt.Sprintf("%x", sha256.Sum256(jsonBytes))
-	resourceSnapshotSpecB := &fleetv1beta1.ResourceSnapshotSpec{
+	resourceSnapshotSpecWithSingleResourceHash := fmt.Sprintf("%x", sha256.Sum256(jsonBytes))
+	resourceSnapshotSpecWithEmptyResource := &fleetv1beta1.ResourceSnapshotSpec{
 		SelectedResources: []fleetv1beta1.ResourceContent{},
 	}
 
-	jsonBytes, err = json.Marshal(resourceSnapshotSpecB)
+	jsonBytes, err = json.Marshal(resourceSnapshotSpecWithEmptyResource)
 	if err != nil {
-		t.Fatalf("failed to create the policy hash: %v", err)
+		t.Fatalf("failed to create the resourceSnapshotSpecWithSingleResourceHash hash: %v", err)
+	}
+	resourceSnapshotSpecWithEmptyResourceHash := fmt.Sprintf("%x", sha256.Sum256(jsonBytes))
+	resourceSnapshotSpecWithMultipleResources := &fleetv1beta1.ResourceSnapshotSpec{
+		SelectedResources: []fleetv1beta1.ResourceContent{serviceResourceContent, secretResourceContent, deploymentResourceContent},
+	}
+	jsonBytes, err = json.Marshal(resourceSnapshotSpecWithMultipleResources)
+	if err != nil {
+		t.Fatalf("failed to create the resourceSnapshotSpecWithMultipleResources hash: %v", err)
+	}
+	resourceSnapshotSpecWithMultipleResourcesHash := fmt.Sprintf("%x", sha256.Sum256(jsonBytes))
+	resourceSnapshotSpecWithTwoResources := &fleetv1beta1.ResourceSnapshotSpec{
+		SelectedResources: []fleetv1beta1.ResourceContent{serviceResourceContent, secretResourceContent},
+	}
+	jsonBytes, err = json.Marshal(resourceSnapshotSpecWithTwoResources)
+	if err != nil {
+		t.Fatalf("failed to create the resourceSnapshotSpecWithMultipleResources hash: %v", err)
+	}
+	resourceSnapshotSpecWithTwoResourcesHash := fmt.Sprintf("%x", sha256.Sum256(jsonBytes))
+	resourceSnapshotSpecSplit1 := resourceSnapshotSpecWithSingleResource
+	resourceSnapshotSpecSplit2 := &fleetv1beta1.ResourceSnapshotSpec{
+		SelectedResources: []fleetv1beta1.ResourceContent{secretResourceContent},
+	}
+	resourceSnapshotSpecSplit3 := &fleetv1beta1.ResourceSnapshotSpec{
+		SelectedResources: []fleetv1beta1.ResourceContent{deploymentResourceContent},
+	}
+	resourceSnapshotSpecSplit4 := &fleetv1beta1.ResourceSnapshotSpec{
+		SelectedResources: []fleetv1beta1.ResourceContent{serviceResourceContent, secretResourceContent},
```

Comment:
forget to post this comment, please use these values directly instead of creating a variable. From the variable name, it's hard to tell what's the usage of them.